pandera/schemas.py

```diff
@@ -689,6 +689,168 @@ def to_yaml(self, fp: Union[str, Path] = None):
+
+        :param keys: list of labels
+        :param drop: bool, default True
+        :param append: bool, default False
+        :return: a new :class:`DataFrameSchema` with specified column(s) in the index.
+        """
+        from pandera.schema_components import Index, MultiIndex
+
+        new_schema = copy.deepcopy(self)
+
+        keys_temp: List = list(keys) if not isinstance(keys, List) else keys
+
+        # ensure all specified keys are present in the columns
+        try:
+            not_in_cols: List[str] = [
+                x for x in keys_temp if x not in new_schema.columns.keys()
+            ]
+            assert not_in_cols == []
+        except AssertionError:
+            raise Exception(f"Keys {not_in_cols} not found in schema columns!")
+
+        # ensure no duplicates
+        try:
+            dup_cols: List[str] = [x for x in set(keys_temp) if keys_temp.count(x) > 1]
+            assert dup_cols == []
+        except AssertionError:
+            raise Exception(f"Keys {dup_cols} are duplicated!")
+
+        # if there is already an index, append or replace according to parameters
+        ind_list: List = (
+            []
+            if new_schema.index is None or not append
+            else list(new_schema.index.columns.values())
+            if isinstance(new_schema.index, MultiIndex) and append
+            else [new_schema.index]
+        )
+
+        for col in keys_temp:
+            ind_list.append(
+                Index(
+                    pandas_dtype=new_schema.columns[col].pandas_dtype,
+                    name=col,
+                    checks=new_schema.columns[col].checks,
+                    nullable=new_schema.columns[col].nullable,
+                    allow_duplicates=new_schema.columns[col].allow_duplicates,
+                    coerce=new_schema.columns[col].coerce,
+                )
+            )
+
+        new_schema.index = (
+            ind_list[0] if len(ind_list) == 1 else MultiIndex(ind_list)
+        )
+
+        # if drop is True as defaulted, drop the columns moved into the index
+        if drop:
+            new_schema = new_schema.remove_columns(keys_temp)
+
+        return new_schema
+
+    def reset_index(self, level: List[str] = None, drop: bool = False):
+        """
+        A method for reseting the :class:`Index` of a :class:`DataFrameSchema`.
+
+        :param level: list of labels
+        :param drop: bool, default True
+        :return: a new :class:`DataFrameSchema` with specified column(s) in the index.
+
+        """
+        from pandera.schema_components import Column, Index, MultiIndex
+
+        new_schema = copy.deepcopy(self)
+
+        try:
+            assert new_schema.index is not None
+        except AssertionError:
+            raise Exception("There is currently no index set for this schema.")
+
+
+        # ensure no duplicates and tuple type
+        level_temp: Union[List[Any], List[str], None] = list(set(level)) if level is not None else []
```

Comment:
I don't think `None` should be an allowable type type, since this'll be an empty list of `level is None`